### ISSUE TITLE
BUG: zip.extractAllToAsync callback not resolving

### ISSUE THREAD

**Issue by leinelissen** [cthackers/adm-zip#407]
Hi there!

Apparently the callback from the `extractAllToAsync` never resolves from v0.5.7. This breaks my flows where I await the result by promise-wrapping the call. I've created a reproduction repo here: https://github.com/leinelissen/adm-zip-async-bug

Cheers,
Lei

**Comment by KirrsGool** [comment]
`extractAllToAsync` has some initialization bug

But if you just want it to call back, you can try this
```js
const AdmZip = require('adm-zip');

const zip = new AdmZip('./unzip-me.zip');

console.log('Starting extraction...');
// use all four paramaters
// bug: if you use three paramaters, it will never call back
zip.extractAllToAsync('./out', true, false, (err) => {
    if (err) console.err('Encountered error while unzipping', err);
    console.log('All files are now unzipped!')
})
```

**Comment by gojevicmario** [comment]
@chlebon i had the same issue. Use the solution above and //@ts-ignore and it should work

**Comment by leinelissen** [comment]
As there is no indication of this being fixed in adm-zip itself, I went ahead and updated the type definitions (https://github.com/DefinitelyTyped/DefinitelyTyped/pull/59070). They have now been merged and released as v0.5.0 of `@types/adm-zip`. This resolves my particular issue, so I'll close this ticket.
